server/cherry_pick.go

```diff
@@ -41,10 +89,18 @@ func (s *Server) handleCherryPick(ctx context.Context, commenter, body string, p
 	if !pr.Merged.Valid || !pr.Merged.Bool {
 		return nil
 	}
-	cmdOut, err := s.doCherryPick(ctx, strings.TrimSpace(args[1]), nil, pr)
-	if err != nil {
-		msg = fmt.Sprintf("Error trying doing the automated Cherry picking. Please do this manually\n\n```\n%s\n```\n", cmdOut)
-		return err
+
+	select {
+	case <-s.cherryPickStopChan:
+		return errors.New("server is closing")
+	case s.cherryPickRequests <- &cherryPickRequest{
```

Comment:
When the server is in shutdown, both of these channels will be closed.  The go runtime considers a write to a closed channel as a valid condition to proceed in a select.

Since select will choose one of these cases at random, this will randomly panic if an event comes in when we're in shutdown.

Reply:
Yeah that's true. It's my fault really, I just somehow ignored it, when I shouldn't have. Thanks for digging into it..

I think we can do a non-blocking read of the stop channel to check if the channel is closed or not. And then move to the next case.

Reply:
Thanks for digging into this @ashishbhate I really appreciate it 🤗 I quite rushed into this to complete before my time off. Eventually it remained after it, but better late than buggy :)